jujugui/static/gui/src/app/components/profile/model-list/model-list.js

```diff
@@ -74,6 +74,29 @@ class ProfileModelList extends React.Component {
     });
   }
 
+  _confirmDestroy(modelUUID) {
+    this.setState({notification: null});
+    this.props.destroyModel(modelUUID, (error, data) => {
+      if (error) {
+        this.props.addNotification({
+          title: 'Error destroying model',
+          message: error,
+          level: 'error'
+        });
+      }
+      if (data != {}) {
```

Comment:
I believe here you're checking to see if the data object returned is not empty. I'm a little confused here as to why an error destroying model wouldn't be returned as an error above. If the `destroyModel` call returns error values in the data then maybe we need to document that, or fix it in the api.

In any event, if you want to check if an object is empty you should use something like `if (Object.keys(data).length) {` as it'll only report keys for enumerable properties.

Reply:
You can have an error from the API call, or models can return an error on destroy (remember that the actual call is destroyModels and takes more than one; the object is a map of model uuids to responses), or potentially both, so we need to display all errors. I'll comment to that effect.